exec-path-from-shell.el

```diff
@@ -188,29 +193,69 @@ shell-escaped, so they may contain $ etc."
 
+(defun exec-path-from-shell-getenvs--nushell (names)
+  "Use nushell to get the value of env vars with the given NAMES.
+
+Execute the shell according to `exec-path-from-shell-arguments'.
+The result is a list of (NAME . VALUE) pairs."
+  (let* ((shell (exec-path-from-shell--shell))
+         (expr (format "[ %s ] | to json"
+                       (string-join
+                        (mapcar (lambda (name)
+                                  (format "$env.%s?" (exec-path-from-shell--double-quote name)))
+                                names)
+                        ", ")))
+         (shell-args (append exec-path-from-shell-arguments (list "-c" expr))))
+    (with-temp-buffer
+      (exec-path-from-shell--debug "Invoking shell %s with args %S" shell shell-args)
+      (let ((exit-code (exec-path-from-shell--warn-duration
+                        (apply #'call-process shell nil t nil shell-args))))
+        (exec-path-from-shell--debug "Shell printed: %S" (buffer-string))
+        (unless (zerop exit-code)
+          (error "Non-zero exit code from shell %s invoked with args %S.  Output was:\n%S"
+                 shell shell-args (buffer-string))))
+      (goto-char (point-min))
+      (let ((json-array-type 'list)
+            (json-null :null))
+        (let ((values (json-read-array))
+              result)
+          (while values
+            (let ((value (car values)))
+              (push (cons (car names)
+                          (unless (eq :null value)
+                            (if (listp value)
+                                (string-join value path-separator)
+                              value)))
+                    result))
+            (setq values (cdr values)
+                  names (cdr names)))
+          result)))))
+
 (defun exec-path-from-shell-getenvs (names)
   "Get the environment variables with NAMES from the user's shell.
 
 Execute the shell according to `exec-path-from-shell-arguments'.
 The result is a list of (NAME . VALUE) pairs."
   (when (file-remote-p default-directory)
     (error "You cannot run exec-path-from-shell from a remote buffer (Tramp, etc.)"))
-  (let* ((random-default (md5 (format "%s%s%s" (emacs-pid) (random) (current-time))))
-         (dollar-names (mapcar (lambda (n) (format "${%s-%s}" n random-default)) names))
-         (values (split-string (exec-path-from-shell-printf
-                                (mapconcat #'identity (make-list (length names) "%s") "\\000")
-                                dollar-names) "\0")))
-    (let (result)
-      (while names
-        (prog1
-            (let ((value (car values)))
-              (push (cons (car names)
-                          (unless (string-equal random-default value)
-                            value))
-                    result))
-          (setq values (cdr values)
-                names (cdr names))))
-      result)))
+  (if (exec-path-from-shell--nushell-p)
```

Comment:
This is the only thing I found that didn't work in the PR. This function seems to take an argument (the shell) but that is missing here. I tried adding the argument in my local copy as `(exec-path-from-shell--shell)` and after that it was working pretty much perfect, including getting additional environment variables from the nushell env.

Reply:
Ah, silly mistake when I tied it together at the last minute then. Thanks for the feedback. I'll fix that issue and merge, then you can see how you get along with the MELPA version.